File: marcenaria-java-activej/src/main/java/github/gmess/models/geometrias/Esfera.java
package github.gmess.models.geometrias;

import github.gmess.utils.NumberUtils;

/**
 * Esfera
 */
public record Esfera(
  String geometria,
  String estrutura,
  String raio
) implements Geometria {
  
  @Override
  public final String getGeometria() {
    return "esfera";
  }

  @Override
  public final String getEstrutura() {
    return this.estrutura;
  }

  @Override
  public final double getArea() {
    final double raio = NumberUtils.centimetersToDouble(this.raio);

    return NumberUtils.FOUR_PI * Math.pow(raio, 2);
  }
 
}

File: MarcenariaDotNet/Dtos/Geometrias/Esfera.cs
using System.Text.Json.Serialization;
using MarcenariaDotNet.Utils;

namespace MarcenariaDotNet.Dtos.Geometrias;

public struct Esfera : Geometria
{
  public String Estrutura { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Raio { get; init; }

    double Geometria.GetArea()
    {
        return NumberUtils.FOUR_PI * Math.Pow(Raio, 2);
    }

    string Geometria.GetEstrutura()
    {
        return Estrutura;
    }

    string Geometria.GetGeometria()
    {
       return "esfera";
    }
}
